src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1879,6 +1885,82 @@ private BoundExpression BindCast(CastExpressionSyntax node, DiagnosticBag diagno
             return BindCastCore(node, operand, targetType, wasCompilerGenerated: operand.WasCompilerGenerated, diagnostics: diagnostics);
         }
 
+        private BoundExpression BindIndexExpression(PrefixUnaryExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            CheckFeatureAvailability(node, MessageID.IDS_FeatureIndexOperator, diagnostics);
+            GetWellKnownTypeMember(Compilation, WellKnownMember.System_Index__ctor, diagnostics, syntax: node);
+
+            return BindUnaryOperator(node, diagnostics);
+        }
+
+        private BoundExpression BindRangeExpression(RangeExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            CheckFeatureAvailability(node, MessageID.IDS_FeatureRangeOperator, diagnostics);
+            TypeSymbol rangeType = GetWellKnownType(WellKnownType.System_Range, diagnostics, node);
+
+            if (!rangeType.IsErrorType())
+            {
+                if (node.LeftOperand is null)
+                {
+                    if (node.RightOperand is null)
+                    {
+                        GetWellKnownTypeMember(Compilation, WellKnownMember.System_Range__All, diagnostics, syntax: node);
+                    }
+                    else
+                    {
+                        GetWellKnownTypeMember(Compilation, WellKnownMember.System_Range__ToEnd, diagnostics, syntax: node);
+                    }
+                }
+                else
+                {
+                    if (node.RightOperand is null)
+                    {
+                        GetWellKnownTypeMember(Compilation, WellKnownMember.System_Range__FromStart, diagnostics, syntax: node);
+                    }
+                    else
+                    {
+                        GetWellKnownTypeMember(Compilation, WellKnownMember.System_Range__Create, diagnostics, syntax: node);
+                    }
+                }
```

Comment:
Consider using the `if` blocks to select the `WellKnownMember` and call `GetWellKnownTypeMember` after the `if`s. And perhaps use `?:` rather than `if`s.